### PR TITLE
add 3Dmol.js v1.0.6 with git auto-update config

### LINKED ISSUE #7940: [Request] Add 3Dmol.js
**Library name:** 3Dmol.js
**Git repository url:** https://github.com/3dmol/3Dmol.js/tree/master/3Dmol
**npm package url(optional):** 
**License(s):**
**Official homepage:**
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by kennynaoh**
PR for #7940 
@pvnr0082t please help me review this pull request, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/3dmol/3Dmol.js
 * Official website (optional, not the repository): http://3dmol.csb.pitt.edu
 * NPM package url (optional):
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 166
   - Count of watchers: 26
   - Count of forks: 45
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #7940, cc @dkoes

### CODE REVIEW

File: ajax/libs/3Dmol/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "webGL"
+  ],
+  "author": "David Koes and contributors",
+  "homepage": "http://3dmol.csb.pitt.edu",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/3dmol/3Dmol.js.git",
+    "basePath": "3Dmol",
+    "files": [
+      "3dmol*"
```
Comment (pvnr0082t) [inline ajax/libs/3Dmol/package.json]: @kennynaoh I think we need to host the files in `release` folder but I'm not sure. Could you confirm it?

**Comment by kennynaoh** [comment]
@pvnr0082t 
I have confirm with author about the files which need to be hosted.
Please help review it again.
Thanks!

**Comment by dkoes** [comment]
Looks good to me.

**Comment by PeterDaveHello** [comment]
@dkoes Thanks.

**Comment by pvnr0082t** [comment]
@PeterDaveHello I think this PR is ok. Please review it again, thank you.

File: ajax/libs/3Dmol/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "webGL"
+  ],
+  "author": "David Koes and contributors",
+  "homepage": "http://3dmol.csb.pitt.edu",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/3dmol/3Dmol.js.git",
+    "basePath": "release",
+    "files": [
+      "3Dmol*"
```
Comment (PeterDaveHello) [inline ajax/libs/3Dmol/package.json]: @kennynaoh why don't use jut use `"**/*"`? Any reason?

Comment (kennynaoh) [inline ajax/libs/3Dmol/package.json]: OK, I will change the format.
Thanks!

**Comment by kennynaoh** [comment]
@pvnr0082t 
I have fixed the `npmFileMap`.
Please help me review it again.
Thanks!

File: ajax/libs/3Dmol/package.json (hunk 2 above)
Comment (PeterDaveHello) [inline ajax/libs/3Dmol/package.json]: This is not about format, it's pattern. So no reason?

**Comment by pvnr0082t** [comment]
@PeterDaveHello I think this PR is ok. Please review it again, thank you.

File: ajax/libs/3Dmol/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "target": "git://github.com/3dmol/3Dmol.js.git",
+    "basePath": "release",
+    "files": [
+      "**/*"
+    ]
+  },
+  "licenses": [
+    "MIT",
+    "LGPL-3.0",
+    "BSD-3-Clause"
```
Comment (PeterDaveHello) [inline ajax/libs/3Dmol/package.json]: Are there so many licenses? @kennynaoh @pvnr0082t

Comment (kennynaoh) [inline ajax/libs/3Dmol/package.json]: I have confirmed with author by this [issue](https://github.com/3dmol/3Dmol.js/issues/177).
Is this ok?
Thanks!

Comment (PeterDaveHello) [inline ajax/libs/3Dmol/package.json]: He only said BSD-3-Clause, so what about the others?

Comment (pvnr0082t) [inline ajax/libs/3Dmol/package.json]: [L1](https://github.com/3dmol/3Dmol.js/blob/master/LICENSE#L1) in `LICENSE` file show __3Dmol.js incorporates code from GLmol__ and [L37](https://github.com/3dmol/3Dmol.js/blob/master/LICENSE#L37) said __GLmol under dual license of MIT or LGPL3__.  

So, do they seperate? I mean that `3Dmol.js` is under BSD-3-Clause only and `GLmol` is under MIT or LGPL3.

Comment (kennynaoh) [inline ajax/libs/3Dmol/package.json]: I saw it in [license](https://github.com/3dmol/3Dmol.js/blob/1.0.5/LICENSE#L37).
Is this correct?
Thanks!

Comment (PeterDaveHello) [inline ajax/libs/3Dmol/package.json]: I only see you added 3Dmol, is GLmol a part of 3Dmol?

Comment (dkoes) [inline ajax/libs/3Dmol/package.json]: 3Dmol.js is a derived work from three.js and GLmol.

Comment (PeterDaveHello) [inline ajax/libs/3Dmol/package.json]: hmmm ... so there are some codes come from three.js and GLmol, so 3Dmol will be released under multiple licenses, am I correct?

Comment (dkoes) [inline ajax/libs/3Dmol/package.json]: Technically, 3Dmol.js is only BSD licensed - the GLmol and three.js parts have been relicensed from MIT to BSD, which according to this chart is acceptable:
https://en.wikipedia.org/wiki/License_compatibility#/media/File:Floss-license-slide-image.png

I include everything in the LICENSE file for the sake of completeness and proper attribution.

Comment (PeterDaveHello) [inline ajax/libs/3Dmol/package.json]: hmmmm ... so I am wondering if we should list all the license from its components ...

Comment (PeterDaveHello) [inline ajax/libs/3Dmol/package.json]: @dkoes do you have any suggestions about the license display in cdnjs for your lib?

Comment (dkoes) [inline ajax/libs/3Dmol/package.json]: BSD-3-Clause

Comment (PeterDaveHello) [inline ajax/libs/3Dmol/package.json]: Thanks @dkoes!
@kennynaoh please send a PR to fix that.

Comment (kennynaoh) [inline ajax/libs/3Dmol/package.json]: No problem!

**Pull request merged**